metricbeat/module/docker/diskio/helper.go

```diff
@@ -174,5 +175,12 @@ func calculatePerSecond(duration time.Duration, old uint64, new uint64) float64
 	if value < 0 {
 		value = 0
 	}
-	return value / duration.Seconds()
+
+	timeSec := duration.Seconds()
+	if timeSec == 0 {
+		logp.Err("duration is zero when calculating per-second values")
+		return -1
```

Comment:
I think that in this case we can assume that if time hasn't passed (duration zero), then the value hasn't changed and we can return zero here.

Returning -1 can lead to weird values in the final events, unless we use this value to skip fields with this value.